### PR TITLE
feat: remove personal info option in request tracking

### LINKED ISSUE #254: Remove the PII information from the request logs automatically
**Is your feature request related to a problem? Please describe.**
The customer requests us to not log PII information in our requests. This is currently supported by excluding headers, but not in the body. 

**Describe the solution you'd like**
I'm looking for something that would also clear out properties from the body (I know content-type wise this might be a not so easy task)

### PR DESCRIPTION
**Pull request by stijnmoreels**
Make the tracked request/response body sanitization avaialable for overriden so custom (possible secret) keys in the body can be removed from tracking.

Closes #254

### CODE REVIEW

File: src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs [hunk 1]
```diff
@@ -269,61 +273,61 @@ private void LogRequest(string requestBody, string responseBody, HttpContext htt
...
         }
 
+        /// <summary>
+        /// Extracts information from the given HTTP <paramref name="requestHeaders"/> to include in the request tracking context.
+        /// </summary>
+        /// <param name="requestHeaders">The headers of the current HTTP request.</param>
+        protected virtual IDictionary<string, StringValues> SanitizeRequestHeaders(IDictionary<string, StringValues> requestHeaders)
+        {
+            return requestHeaders.Where(header => Options.OmittedHeaderNames?.Contains(header.Key) == false);
+        }
```
[the remark this answers is not in the document]
Comment (fgheysels) [reply src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs:283]: I was just wondering why we choose to specify the headers that must be omitted as an array in the options, and why we have to define the headers that must be sanitized by overriding a method.
Maybe I'm overlooking somehting, or am I not aware of something ?

[the remark this answers is not in the document]
Comment (stijnmoreels) [reply src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs:283]: The sanitizing process can also include some custom code. Like removing stuff from the header values. In that case, it seemed better to do that in a method than to pass along a function in the options.

File: src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs [hunk 2]
```diff
@@ -269,61 +273,61 @@ private void LogRequest(string requestBody, string responseBody, HttpContext htt
             return new Dictionary<string, StringValues>();
         }
 
+        /// <summary>
+        /// Extracts information from the given HTTP <paramref name="requestHeaders"/> to include in the request tracking context.
```
[the remark this answers is not in the document]
Comment (fgheysels) [reply src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs:277]: I have modified the xml doc

File: src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs [hunk 3]
```diff
@@ -269,61 +273,61 @@ private void LogRequest(string requestBody, string responseBody, HttpContext htt
...
+        /// </summary>
+        /// <param name="requestHeaders">The headers of the current HTTP request.</param>
```
[the remark this answers is not in the document]
Comment (fgheysels) [reply src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs:279]: I'd add something like this (given the assumption that we need to do the sanitization in this method as well)

```
/// <remarks>Override this method if there are headers that contain sensitive information that should not be logged via request-tracking.</remarks.
```

I'd also add a returns section
```
/// <returns>A collection of headers and the header contents that must be logged via request-tracking
```

**Review by fgheysels — APPROVED** [review]

**Pull request merged**